hil-test/tests/delay_async.rs

```diff
@@ -0,0 +1,167 @@
+//! Async Delay Test
+//!
+//! Specifically tests the various implementations of the
+//! `embedded_hal_async::delay::DelayNs` trait.
+
+//% CHIPS: esp32 esp32c2 esp32c3 esp32c6 esp32h2 esp32s2 esp32s3
+
+#![no_std]
+#![no_main]
+
+use embedded_hal_async::delay::DelayNs;
+#[cfg(systimer)]
+use esp_hal::timer::systimer::{Alarm, FrozenUnit, SystemTimer};
+use esp_hal::{peripherals::Peripherals, timer::timg::TimerGroup};
+use hil_test as _;
+
+struct Context {
+    peripherals: Peripherals,
+}
+
+async fn test_async_delay_ns(mut timer: impl DelayNs, duration: u32) {
```

Comment:
Mind adding a loop in here to ensure repeated calls to `delay_ns` work as expected.
It happens quite often in drivers that the first one works but the next ones don't.